### DISCUSSION TITLE
Peacock - No module named 'PyQt5' Error

### DISCUSSION THREAD

**Discussion by Josh-Web-700** [idaholab/moose#26918] [Q&A General]
I have downloaded MOOSE and am working on my first example simulations, but visualizing them with Peacock only leads to the following error:

`(base) scholar2bee@MSI:~$ conda activate peacock`
`(peacock) scholar2bee@MSI:~$ docker run -it -v projects:/projects idaholab/moose-dev:latest`
`bash-4.4# cd /projects/babbler/problems`
`bash-4.4# /projects/moose/python/peacock/peacock -r pressure_diffusion_out.e`
`Traceback (most recent call last):`
 ` File "/projects/moose/python/peacock/peacock", line 36, in <module>`
 `   run_peacock()`
 ` File "/projects/moose/python/peacock/peacock", line 28, in run_peacock`
 `   sys.exit(main(sys.argv))`
 ` File "/projects/moose/python/peacock/peacock", line 12, in main`
 `   from peacock.CheckRequirements import has_requirements`
 ` File "/projects/moose/python/peacock/__init__.py", line 11, in <module>`
 `   from . import base`
 ` File "/projects/moose/python/peacock/base/__init__.py", line 10, in <module>`
 `   from .MooseWidget import MooseWidget`
 ` File "/projects/moose/python/peacock/base/MooseWidget.py", line 10, in <module>`
 `   from PyQt5 import QtCore`
`ModuleNotFoundError: No module named 'PyQt5'`

I am not sure how to move forward without being able to visualize the simulations in Peacock. What is the best way to fix this?

**Comment by milljm** [comment]
As soon as you executed `docker`, you exited the environment capable of running Peacock.

I don't think Docker is capable of looping back out of it (its a Container), to display/connect to your host systems Graphics API. Er... it may be possible, but I don't know the steps you'll need to take in order to do so.

If you want to view your simulation results, I suggest using Paraview. Basically you would use Docker to run the simulation, and then on the host machine open the result in Paraview.

**Reply by milljm** [reply]
I converted this to a discussion, as this is not a bug of MOOSE.
